### PR TITLE
API for rotations produced by LookBehavior

### PR DESCRIPTION
**Pull request by ZeroMemes**
Fully deterministic with support for simulation of future ticks.

This PR does modify the way that RotationMoveEvent works in coordination with LookBehavior, meaning that randomLooking and simulated mouse input does have an affect on player movement. However, Baritone for 1.13+ already has affected movement because of randomLooking113.

### CODE REVIEW

**Comment by ZacSharp** [comment]
Does this mean `randomLooking113` can be moved into the normal rotation processing once this is merged?

File: src/api/java/baritone/api/behavior/look/IAimProcessor.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+public interface IAimProcessor {
+
+    /**
+     * Returns the actual rotation that will be used when the desired rotation is requested. This is not guaranteed to
+     * return the same value for a given input.
+     *
+     * @param desired The desired rotation to set
+     * @return The actual rotation
+     */
+    Rotation nextRotation(Rotation desired);
```
Comment (leijurv) [inline src/api/java/baritone/api/behavior/look/IAimProcessor.java:34]: I think this javadoc should explicitly say whether or not calling this function will advance the RNG by one tick, or not

**Review by leijurv — APPROVED** [review]

File: src/api/java/baritone/api/behavior/look/IAimProcessor.java (hunk 1 above)
Comment (ZacSharp) [reply src/api/java/baritone/api/behavior/look/IAimProcessor.java:34]: I think using one method to advance by one tick *or* peek the next tick depending on the instance is not good.
Can `IAimProcessor` maybe be split into one interface which can only be peeked and forked and a second interface which extends the first one by adding methods to tick it and to get a rotation and advance one tick simultaneously? The root processor would implement just the first interface, the forked ones the second interface.

Comment (ZeroMemes) [reply src/api/java/baritone/api/behavior/look/IAimProcessor.java:34]: Agreed

**Review by ZacSharp — APPROVED** [review]

**Pull request merged**